cmd/server-main.go

```diff
@@ -495,7 +504,33 @@ func getServerListenAddrs() []string {
 }
 
 // serverMain handler called for 'minio server' command.
-func serverMain(ctx *cli.Context) {
+func serverMain(ctx *cli.Context) error {
+	if ctx.Int("numa-cpunode") > -1 {
+		cpuNode := ctx.Int("numa-cpunode")
+		cpumask, err := numa.NodeToCPUMask(cpuNode)
+		if err != nil {
+			return err
+		}
+
+		if err = numa.SetSchedAffinity(0, cpumask); err != nil {
+			return err
+		}
+
```

Comment:
GOMAXPROCS should be set appropriately to not overprovision threads for cores.
```suggestion
		runtime.GOMAXPROCS(cpumask.OnesCount())
			
```
I seems like this should correctly be set to the logical core count of the CPU.

Reply:
The reason this is not needed @klauspost is that the `SetSchedAffinity` automatically handles this via syscall.Exec() - go automatically then gets the right "get_schedaffinity" value. 

I observed this via `strace` - thats why this kind of hack is needed. Otherwise Go runtime picks the affinity very early on and its impossible to set it correctly.

But I could this as a safety measure.